UI/EmuScreen.cpp

```diff
@@ -1302,11 +1302,23 @@ void EmuScreen::renderUI() {
 }
 
 void EmuScreen::autoLoad() {
+	int autoSlot = -1;
+
 	//check if save state has save, if so, load
-	int lastSlot = SaveState::GetNewestSlot(gamePath_);
-	if (g_Config.bEnableAutoLoad && lastSlot != -1) {
-		SaveState::LoadSlot(gamePath_, lastSlot, &AfterSaveStateAction);
-		g_Config.iCurrentStateSlot = lastSlot;
+	switch (g_Config.iEnableAutoLoad) {
+		case 0 : return; // "AutoLoad Off"
```

Comment:
In general for switches, we use the convention:

```c++
switch (variable) {
case value1:
	stuff();
	break;

case value2:
case value3:
	stuffFor2Or3();

default:
	break;
}
```

It's fine on the same line, but the cases shouldn't indent (like labels) and shoudln't have a space before the `:`.

All that said, maybe it could just be like...

```c++
switch (g_Config.iEnableAutoLoad) {
case (int)SaveStateAutoLoad::OFF: return;

case (int)SaveStateAutoLoad::OLDEST: autoSlot = SaveState::GetOldestSlot(gamePath_); break;
case (int)SaveStateAutoLoad::NEWEST: autoSlot = SaveState::GetNewestSlot(gamePath_); break;

default:
	// The only other values supported are actual slot numbers.
	// (code for that)
}
```

That way it automatically handles if we add new slots later or whatever.

-[Unknown]

Reply:
The `enum` improves readability but makes the config file less intuitive.  I guess users aren't meant to be editing it by hand so no matter.

And as you say, it more easily allows for more slots in future.  With that in mind I spent some time looking into generating
`static const char *autoLoadSaveStateChoices[] = { "Off", "Oldest Save", "Newest Save", "Slot 1", "Slot 2", "Slot 3", "Slot 4", "Slot 5" };`
dynamically based on the
`static const int NUM_SLOTS = 5;`
so there's just that 1 constant to change, but as `PopupMultiChoice` takes a `const` array it can't be done, so far as I can tell.

Reply:
Right - I think it's okay for now.  But `PopupMultiChoiceDynamic` exists for that purpose.

-[Unknown]

Reply:
I did see that and thought I'd cracked it, but didn't think it was going to be suitable for some reason.  Likely I've misunderstood the prototype.  Will have a another look at that for completeness.